src/unused_finder/mod.rs

```diff
@@ -194,11 +194,20 @@ pub fn find_unused_items(
         }
     }
 
-    let unused_files = BTreeMap::from_iter(graph.files.iter().filter(|f| !f.1.is_used));
+    let unused_files = BTreeMap::from_iter(
+        graph
+            .files
+            .drain()
+            .filter(|f| !f.1.is_used && !f.1.import_export_info.allow_unused),
+    );
     let results: Vec<String> = unused_files
         .iter()
         .map(|f| format!("\"{}\",", f.0))
         .chain(graph.files.iter().filter_map(|(path, file)| {
+            // shortcircuit to not report unused items on files with `@allow-unused-file` comment
+            if file.import_export_info.allow_unused {
+                return None;
```

Comment:
should error here if allow_unused is set in on a file with no unused exports.

consider:
```rust
let items = file
    .unused_exports
    .iter()
    .map(|items| items.to_string())
    .collect::<Vec<_>>();

let allow_unused = file.import_export_info.allow_unused 

return if allow_unused && items.is_empty() {
  Some(format!("From file \"{path}\": contains @allow-unused-file annotation with no unused exports. The annotation should be removed."))
} else if !allow_unused && !items.is_empty() { 
  Some(format!("From file \"{}\": {}", path, items))
} else {
  None
}
```